include/aspect/material_model/interface.h

```diff
@@ -797,6 +797,35 @@ namespace aspect
     };
 
 
+    /**
+     * Additional output fields for prescribed field outputs to be added to
+     * the MaterialModel::MaterialModelOutputs structure and filled in the
+     * MaterialModel::Interface::evaluate() function.
+     *
+     * If the advection scheme "prescribed field" is selected, then these
+     * material model outputs will be interpolated onto the corresponding
+     * compositional field.
+     */
+    template <int dim>
+    class PrescribedFieldOutputs : public NamedAdditionalMaterialOutputs<dim>
+    {
+      public:
+        PrescribedFieldOutputs (const unsigned int n_points,
+                                const unsigned int n_comp);
+
+        virtual std::vector<double> get_nth_output(const unsigned int idx) const;
+
+        /**
+         * Prescribed field outputs for all compositional fields at the evaluation points
+         * that are passed to the instance of MaterialModel::Interface::evaluate()
+         * that fills the current object.
+         * copy_properties[q][c] is the prescribed field output at the evaluation point q
+         * for the compositional field with the index c.
```

Comment:
Do we really want n entries here even if we have less than n prescribed fields? This might be confusing because you have to make sure to fill the correct entry. Otherwise your values will be silently ignored.

Reply:
I thought using as many entries as we have compositional fields makes sense because then it is easy to assign values for the correct field in the material model by using `this->introspection().compositional_index_for_name("field_name")`. If we choose a smaller number, we would have to do some conversion between the indices somewhere. 

And if you happen to put your values into the wrong field (and not fill the field you wanted), it will try and copy quietNaNs into the prescribed field (which is what the PrescribedFieldOutputs are initialized to), so the user should notice something is wrong. 

The only problem you get is if you fill the outputs in the material model, but then do not actually use prescribed fields. But that is the same for other plugins and additional outputs, you can fill reaction rates, but if you do not use operator splitting, they will just be ignored. 
And I do not see a good way to check if all fields that are not prescribed fields still have their original quietNaN values in the PrescribedFieldOutputs.

Reply:
> And if you happen to put your values into the wrong field (and not fill the field you wanted), it will try and copy quietNaNs into the prescribed field

They are quiet, so no error will happen, right? How do you notice? Can we add a ``is_finite(vector.max())`` assert or something?

Question: can you visualize this additional output with the postprocessor or does this break because of the NaNs?

Reply:
oh, and can you leave a note in this class (or here) that this is _all_ compositional fields not only the prescribed ones?

Reply:
> Question: can you visualize this additional output with the postprocessor or does this break because of the NaNs?

At the moment, if you don't fill the outputs, it will copy NaNs into your prescribed compositional field, so even without the postprocessor, if you ever look at the output, you will notice it's all NaN. But you can also use the postprocessor to look at the prescribed field outputs, and they will also all be NaNs. So it should be very obvious that something is wrong. 

I am adding an Assert (so it's only triggered in debug mode) that will check the entries that are used are finite, and in release mode, it will just copy NaNs into the fields as described above.